### PR TITLE
Enable eslint rule no-useless-escape

### PR DESCRIPTION
**Pull request by jakebailey**
Follow-up to #54693.

This actually caught a small number of oopsies where we wrote `\r\p` but meant `\r\np`

### CODE REVIEW

**Review by DanielRosenwasser — APPROVED** [review]
I hated reviewing this.

**Comment by jakebailey** [comment]
Sending it was slightly better; at least I had a quick fix but no `--fix` because there are _two_ quick fixes, the "preserve behavior" button or the "actually I meant `\\p`" button.

**Review by sheetalkamat — APPROVED** [review]

File: src/testRunner/unittests/tsserver/cachingFileSystemInformation.ts [hunk 1]
```diff
@@ -397,7 +397,7 @@ describe("unittests:: tsserver:: CachingFileSystemInformation:: tsserverProjectS
     "typescript"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo "Error: no test specified" && exit 1"
```
Comment (jakebailey) [inline src/testRunner/unittests/tsserver/cachingFileSystemInformation.ts:400]: Here's one I got wrong!

Comment (jakebailey) [reply src/testRunner/unittests/tsserver/cachingFileSystemInformation.ts]: Fixed now. Didn't actually affect the test, apparently.

File: src/testRunner/unittests/base64.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ describe("unittests:: base64", () => {
                 "日本語",
                 "🐱",
                 "\x00\x01",
-                "\t\n\r\\\"\'\u0062",
+                "\t\n\r\\\"'\u0062",
```
Comment (jakebailey) [inline src/testRunner/unittests/base64.ts:13]: On second look, I'm not sure if this intended to write `'` or `\\'`. But, I'm not sure what "escaping" there is to check when roundtripping base64 anyway...

File: src/testRunner/unittests/convertToBase64.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ describe("unittests:: convertToBase64", () => {
         });
 
         it("Converts escape sequences correctly", () => {
-            runTest("\t\n\r\\\"\'\u0062");
+            runTest("\t\n\r\\\"'\u0062");
```
Comment (jakebailey) [inline src/testRunner/unittests/convertToBase64.ts:16]: Similarly here.

**Pull request merged**